packages/core/chrome/core-chrome-browser/src/contracts.ts

```diff
@@ -161,4 +162,26 @@ export interface ChromeStart {
    * Get an observable of the current style type of the chrome.
    */
   getChromeStyle$(): Observable<ChromeStyle>;
+  /**
+   * Configuration for serverless projects
+   */
+  project: {
+    /**
+     * Sets the project navigation config to be used for rendering project navigation.
+     * It is used for default project sidenav, project breadcrumbs, tracking active deep link.
+     * @param projectNavigation The project navigation config
+     *
+     * @remarks Has no effect if the chrome style is not `project`.
+     */
+    setNavigation(projectNavigation: ChromeProjectNavigation): void;
+
+    /**
+     * Set custom project sidenav component to be used instead of the default project sidenav.
+     * @param getter A function returning a CustomNavigationComponent.
+     * This component will receive Chrome navigation state as props (not yet implemented)
+     *
+     * @remarks Has no effect if the chrome style is not `project`.
+     */
+    setNavigationComponent(getter: () => CustomNavigationComponent): void;
```

Comment:
Why should this accept a getter instead of a React component? Is there a good reason to do compromising a bit cleaner API?

Reply:
I had some issue initially with `rxjs` and `useObservable`. Fixed in https://github.com/elastic/kibana/pull/156205/commits/6f792267ada44a753de4bb5356cad7a13ebb024d, we now can pass the SideNav component without getter

```ts
chrome.project.setSideNavComponent(MyNav);
```